docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -243,46 +237,120 @@ These common scenarios trigger a violation from MultipleMassSchemaLookupRule.
 |				 |`Schema.describeSObjects(...)`																										|
 
-#### Interpreting MultipleMassSchemaLookupRule Results
+#### Interpreting AvoidMultipleMassSchemaLookups Results
 Match any violation message that you receive with these cases to understand more about the violation.
 
 ##### Loop Case
 
 > `Schema.getGlobalDescribe` was called inside a loop. `[ForEachStatement at AuraEnabledFls:27]`
 
-*Explanation*
+Explanation:
 
 Your code calls `Schema.getGlobalDescribe()` or `Schema.describeSObjects(...)` inside a loop statement. Modify your code to move the `Schema.getGlobalDescribe()` or `Schema.describeSObjects(...)` outside the loop, then rescan your code. 
 
 ##### Multiple Schema Lookups Are Invoked Case
 
 > Multiple expensive schema lookups are invoked. `[Schema.describeSObjects at AuraEnabledFls:27]`
 
-*Explanation*
+Explanation:
 
 Your code invokes `Schema.getGlobalDescribe()` preceded by `Schema.describeSObjects`. Modify your code so that only one expensive schema lookup is invoked.
 
 ##### More Than One Execution in a Path Case
 
 > `Schema.getGlobalDescribe` executed multiple times in the call stack. `[getFields at AuraEnabledFls:27, getFields at AuraEnabledFls:28, getFields at AuraEnabledFls:29]`
 
-*Explanation*
+Explanation:
 
 `Schema.getGlobalDescribe` or `Schema.describeSObjects` is executed multiple times in a single path. Reduce the execution of the method to one time, then rescan your code.
 
-### UnusedMethodRule
+### UnimplementedTypeRule
 
-UnusedMethodRule is a path-based analysis rule that detects many methods contained in your code that aren’t invoked from any entry points that Graph Engine recognizes. UnusedMethodRule detects methods contained in your code that aren’t invoked. It detects:
+UnimplementedTypeRule detects abstract classes and interfaces that are non-global and missing implementations or extensions.
 
-- static methods
-- instance methods
+#### Definition
 
-To invoke the path-based rules run: `scanner:run --engine sfge --projectdir MyDirectory`. 
+UnimplementedTypeRule is a traditional static analysis rule where a violation occurs at a point in the code where the interface or abstract class is declared. It doesn’t use sources or sinks.
+
+#### Interpreting UnimplementedTypeRule Results
+
+Match any violation message that you receive with this case to understand more about the violation.
+
+*Common Case*
+
+> Extend, implement, or delete %s %s
+
+Explanation:
+
+Because this abstract class or interface has no implementations or extensions, it can’t be instantiated. It’s unnecessary and can be deleted.
+
+#### UnimplementedTypeRule Limitations
+
+Because UnimplementedType rule excludes `global` scoped classes from consideration, these classes are prevented from being thrown as false positives and aren’t false negatives.
+
+## Pilot Rules
 
-**Example**: 
-```sfdx scanner:run --engine sfge --projectdir /project/dir --target /project/dir/target1```
+### AvoidDatabaseOperationInLoop
```

Comment:
Anchor is needed.
`### AvoidDatabaseOperationInLoop <a name='AvoidDatabaseOperationInLoop'>#</a>`

Reply:
Again, I globally added anchors. Thanks.